### PR TITLE
feature: added patched needed to support stream balancer_by_lua_*.

### PR DESCRIPTION
**Pull request by dndx**
I hereby granted the copyright of the changes in this pull request
to the authors of this openresty project.

### CODE REVIEW

File: patches/nginx-1.13.3-stream_proxy_timeout_fields.patch [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+ typedef ngx_int_t (*ngx_stream_filter_pt)(ngx_stream_session_t *s,
+diff --git a/src/stream/ngx_stream_proxy_module.c b/src/stream/ngx_stream_proxy_module.c
+index 0afde1c..f197c83 100644
+--- a/src/stream/ngx_stream_proxy_module.c
++++ b/src/stream/ngx_stream_proxy_module.c
+@@ -359,6 +359,7 @@ ngx_stream_proxy_handler(ngx_stream_session_t *s)
+     ngx_stream_proxy_srv_conf_t      *pscf;
+     ngx_stream_upstream_srv_conf_t   *uscf, **uscfp;
+     ngx_stream_upstream_main_conf_t  *umcf;
++    ngx_stream_proxy_ctx_t           *sctx;
```
Comment (agentzh) [inline patches/nginx-1.13.3-stream_proxy_timeout_fields.patch]: Why don't we just use `ctx` here?

File: patches/nginx-1.13.3-stream_proxy_timeout_fields.patch [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
++++ b/src/stream/ngx_stream.h
+@@ -241,6 +241,14 @@ typedef struct {
+ } ngx_stream_module_t;
+ 
+ 
++typedef struct {
++    ngx_msec_t                       connect_timeout;
++    ngx_msec_t                       timeout;
++} ngx_stream_proxy_ctx_t;
++
```
Comment (agentzh) [inline patches/nginx-1.13.3-stream_proxy_timeout_fields.patch]: Style: need 2 blank lines here.

File: patches/nginx-1.13.3-stream_balancer_export.patch [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+--- a/src/stream/ngx_stream_upstream_round_robin.h
++++ b/src/stream/ngx_stream_upstream_round_robin.h
+@@ -141,6 +141,13 @@ ngx_int_t ngx_stream_upstream_get_round_robin_peer(ngx_peer_connection_t *pc,
+     void *data);
+ void ngx_stream_upstream_free_round_robin_peer(ngx_peer_connection_t *pc,
+     void *data, ngx_uint_t state);
++ngx_int_t ngx_stream_upstream_set_round_robin_peer_session(
++    ngx_peer_connection_t *pc, void *data);
++void ngx_stream_upstream_save_round_robin_peer_session(
++    ngx_peer_connection_t *pc, void *data);
```
Comment (agentzh) [inline patches/nginx-1.13.3-stream_balancer_export.patch]: These 2 functions were originally protected by the macro `NGX_STREAM_SSL` and we should keep that in the header file as well.

File: patches/nginx-1.13.3-stream_proxy_timeout_fields.patch (hunk 1 above)
Comment (dndx) [reply patches/nginx-1.13.3-stream_proxy_timeout_fields.patch]: `ngx_stream_proxy_handler` already defines `ctx` above.

Comment (agentzh) [reply patches/nginx-1.13.3-stream_proxy_timeout_fields.patch]: @dndx I see. Then we'd better name it to `pctx` instead of `sctx`.

**Comment by agentzh** [comment]
@dndx Merged. Thanks!

**Pull request closed without merging**